sirepo/package_data/static/js/sirepo-plotting.js

```diff
@@ -3052,992 +3071,110 @@ SIREPO.app.directive('particle', function(plotting, layoutService, utilities) {
-                        }
-                    },
-                    z: {
-                        pos: $(zaxis).position(),
-                        size: {
-                            width: $(zaxis).width(),
-                            height: $(zaxis).height()
-                        }
-                    }
-                };
-                return g;
+        // horizontal line
+        if(startY == endY) {
+            return {
+                left: [bounds.left, startY],
+                top: [-Infinity, bounds.top],
+                right: [bounds.right, endY],
+                bottom: [Infinity, bounds.bottom]
             };
-
-            // the values below take rotation into account
-            $scope.labelGeometry = function() {
-                var xlabel = d3.select('.vtk-info-overlay .x-axis-label');
-                var ylabel = d3.select('.vtk-info-overlay .y-axis-label');
-                var zlabel = d3.select('.vtk-info-overlay .z-axis-label');
-                var xfs = utilities.fontSizeFromString(xlabel.style('font-size'));
-                var yfs = utilities.fontSizeFromString(ylabel.style('font-size'));
-                var zfs = utilities.fontSizeFromString(zlabel.style('font-size'));
-                var g = {
-                    x: {
-                        pos: {
-                            top: parseInt(xlabel.attr('y') || '0'),
-                            left: parseInt(xlabel.attr('x') || '0')
-                        },
-                        size: {
-                            width: xlabel.text().length * xfs,
-                            height: xfs
-                        }
-                    },
-                    y: {
-                        pos: {
-                            top: parseInt(ylabel.attr('x') || '0'),
-                            left: parseInt(ylabel.attr('y') || '0')
-                        },
-                        size: {
-                            width: yfs,
-                            height: ylabel.text().length * yfs
-                        }
-                    },
-                    z: {
-                        pos: {
-                            top: parseInt(zlabel.attr('x')),
-                            left: parseInt(zlabel.attr('y'))
-                        },
-                        size: {
-                            width: zfs,
-                            height: zlabel.text().length * zfs
-                        }
-                    }
-                };
-                return g;
+        }
+        // vertical line
+        if(startX == endX) {
+            return {
+                left: [bounds.left, -Infinity],
+                top: [startX, bounds.top],
+                right: [bounds.right, Infinity],
+                bottom: [endX, bounds.bottom]
             };
+        }
 
+        var m = (endY - startY) / (endX - startX);
+        return {
+            left: [bounds.left, startY + m * (bounds.left - startX)],
```

Comment:
the math here is repetitive. It relates to the above comment. Is there a way to modularize the tests in a loop. It seems we maybe want to be using a 2d matrix (dims) not an object for this. That would at least allow a loop.